prediction_market_agent_tooling/markets/seer/data_models.py

```
@@ -30,21 +47,174 @@ class CreateCategoricalMarketsParams(BaseModel):
+    @property
+    def has_valid_answer(self) -> bool:
+        # We assume that, for the market to be resolved as invalid, it must have both:
+        # 1. An invalid outcome AND
+        # 2. Invalid payoutNumerator is 1.
+
+        try:
+            self.outcome_as_enums[SeerOutcomeEnum.NEUTRAL]
+        except KeyError:
+            raise ValueError(
+                f"Market {self.id} has no invalid outcome. {self.outcomes}"
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)